libraries/pet_service/pet_service.lua

```diff
@@ -0,0 +1,176 @@
+        available_frames    = {struct.bitfield(4)},
+        available_attach    = {struct.bitfield(32)},
+        name                = {struct.string(0x10)},
+        hp                  = {struct.uint16},
+        hp_max              = {struct.uint16},
+        mp                  = {struct.uint16},
+        mp_max              = {struct.uint16},
+        melee               = {struct.uint16},
+        melee_max           = {struct.uint16},
+        ranged              = {struct.uint16},
+        ranged_max          = {struct.uint16},
+        magic               = {struct.uint16},
+        magic_max           = {struct.uint16},
+        str                 = {struct.uint16},
+        str_modifier        = {struct.uint16},
+        dex                 = {struct.uint16},
+        dex_modifier        = {struct.uint16},
+        vit                 = {struct.uint16},
+        vit_modifier        = {struct.uint16},
+        agi                 = {struct.uint16},
+        agi_modifier        = {struct.uint16},
+        int                 = {struct.uint16},
+        int_modifier        = {struct.uint16},
+        mnd                 = {struct.uint16},
+        mnd_modifier        = {struct.uint16},
+        chr                 = {struct.uint16},
+        chr_modifier        = {struct.uint16},
+    })}
+}))
+
+packet.incoming:register_init({
+    [{0x037}] = function(p) -- While this packet is mostly player data, it does occassionally update the pet index when no other pet related packet is sent. For example, when moving into zones where the pet is supressed, such as cities and towns, this packet will set the pet index to 0.
+        data.index = p.pet_index
+        if p.pet_index and p.pet_index ~= 0 then
+            data.active = true
+        else
+            data.active = false
+        end
+    end,
+    [{0x067}] = function(p)
+        if p.type == 4 then
+            data.index = p.pet_index
+            data.id = p.pet_id
+            data.owner_index = p.owner_index
+            data.hp_percent = p.hp_percent
+            data.mp_percent = p.mp_percent
+            data.tp = p.pet_tp
+            if p.pet_index and p.pet_index ~= 0 then
+                data.active = true
+            else
+                data.active = false
+            end
+        end
+    end,
+    [{0x068}] = function(p)
+        if p.type == 4 then
+            data.owner_index = p.owner_index
+            data.owner_id = p.owner_id
+            data.index = p.pet_index
+            data.hp_percent = p.hp_percent
+            data.mp_percent = p.mp_percent
+            data.tp = p.pet_tp
+            data.target_id = p.target_id
+            data.name = p.pet_name
+            if p.pet_index and p.pet_index ~= 0 then
+                data.active = true
+            else
+                data.active = false
+            end
+        end
+    end,
+    [{0x044,0x12}] = function(p)
+
+        data.automaton.head.raw         = p.automaton_head
```

Comment:
As mentioned above, I would remove all `raw` values and just adjust it here for the item ID. Should also unalign the assignments, keep it consistent with other files and other code segments within this file.